src/modules/file/file.controller.ts

```diff
@@ -213,6 +213,9 @@ export class FileController {
     @Body() updateFileMetaDto: UpdateFileMetaDto,
     @Client() clientId: string,
   ) {
+    if (!updateFileMetaDto) {
+      throw new BadRequestException('Missing update file metadata');
+    }
```

Comment:
A small tip, you can use Nestjs tools to strip non desired values from the body of the request, allowing you to check if at least one of the fields is present by using `Object.keys(updateFileMetaDto).length === 0 `

  ```
@UsePipes(
    new ValidationPipe({
      whitelist: true,
    }),
  )
  async updateFileMetadata(
    @UserDecorator() user: User,
    @Param('uuid', ValidateUUIDPipe)
    fileUuid: File['uuid'],
    @Body() updateFileMetaDto: UpdateFileMetaDto,
    @Client() clientId: string,
  ) {
    if (!updateFileMetaDto || Object.keys(updateFileMetaDto).length === 0) {
      throw new BadRequestException('Missing update file metadata');
    }
```

Reply:
I have moved this to the service as its critical for the good functioning of the usecase by itself

Reply:
That is not correct @larryrider. The responsibility of the controller is precisely to validate the input of the request before passing it to the service/usecase. 

If you have business rules regarding to what can be updated, what not and in which situation, use the File entity to add the business rules to the domain, so you keep those details where they should be. 

Otherwise, a rule that is bound to the file and should be enforced always, it's enforced per each use case, increasing the margin to miss that detail in other usecases. 

Then, we can just call that method (like `isValidPlainName()`) on the beginning of the use case method.